server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java

```diff
@@ -104,11 +106,13 @@ public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        ReleasableBytesReference source = request.requiredReleasableContent();
+        source.mustIncRef();
```

Comment:
By acquiring a ref here we need to account for the possibility of an exception in the code that copies the params into `indexRequest` - for instance `paramAsTime` can throw an IAE if the parameter cannot be parsed.

Pretty sure that all we need to do for that is to move this `mustIncRef()` call to happen just after the `return` - that way if this method completes exceptionally then we never acquired a ref to clean up. The instantiation of the lambda in the `return` line cannot throw, right?

Reply:
Thanks for pointing out. Lambda should not throw as far as I can tell following code, if it throws it should notify listener and close reference. Moving incRef into lambda.